src/tests/dotnet-monitor/RemoteTestExecution.cs

```diff
@@ -31,47 +37,30 @@ private void SendSignal()
         {
             //We cannot use named synchronization primitives since they do not work across processes
             //on Linux. Use redirected standard input instead.
-            RemoteProcess.Process.StandardInput.Write('0');
-            RemoteProcess.Process.StandardInput.Flush();
+            TestRunner.StandardInput.Write('0');
+            TestRunner.StandardInput.Flush();
         }
 
-
-        public void Dispose()
+        public static RemoteTestExecution StartRemoteProcess(string loggerCategory, ITestOutputHelper outputHelper)
         {
-            SendSignal();
+            TestRunner runner = new TestRunner(CommonHelper.GetTraceePath("EventPipeTracee") + " " + loggerCategory,
+                outputHelper, redirectError: true, redirectInput: true);
+            runner.Start();
+
+            ReadAllOutput(runner.StandardOutput, runner.StandardError, outputHelper);
```

Comment:
I know that the previous code didn't wait for the std out and std err streams to finish processing, but I think that would be helpful for the disposal of the RemoteTestExecution object to wait on it so that std out and std err are fully dumped before the test finishes executing.

Reply:
That makes sense, I can add this.